### DISCUSSION TITLE
HxGrid Inline Editing ValueChanged on HxInputNumber

### DISCUSSION THREAD

**Discussion by bluisanajmr** [havit/Havit.Blazor#755] [Q&A] [answered]
I have an HxGrid with inline editing.  Everything works great up until the point where I wanted to create a valuechanged event for one of the inline HxInputNumber fields. 

```
<HxGridColumn HeaderText="Quantity">
    <ItemTemplate Context="InvoiceLineItemContext">
        @if (edititem == InvoiceLineItemContext)
        {
            <HxInputNumber TValue="Decimal" Value="@InvoiceLineItemContext.Quantity" ValueExpression="() => InvoiceLineItemContext.Quantity" ValueChanged="calculatequantity" Decimals="2" />
        }
        else
        {
            <div>@InvoiceLineItemContext.Quantity.ToString("c0")</div>
        }
    </ItemTemplate>
</HxGridColumn>

```
All of my other fields are using a simple bind like this. 

```
 <HxGridColumn HeaderText="Price">
     <ItemTemplate Context="InvoiceLineItemContext">
         @if (edititem == InvoiceLineItemContext)
         {
             <HxInputNumber @bind-Value="InvoiceLineItemContext.Price" Decimals="2" InputGroupStartText="$" />
         }
         else
         {
             <div>@InvoiceLineItemContext.Price.ToString("c0")</div>
         }
     </ItemTemplate>
 </HxGridColumn>
```

The quantity inputnumber field renders but I am unable to get a value to bind in it.  Any number that I enter into the inputnumber field changes back to 0 as soon as I move out of the field.

**Comment by hakenr** [comment, marked as answer]
How does your `calculatequantity` method look like? You have to save the value to the model there.
In this setup I would strongly recommend using `bind-Value="InvoiceLineItemContext.Quantity" bind-Value:after="calculatequantity"` instead of `ValueChanged`.

**Comment by bluisanajmr** [comment]
Thanks Hadenr,  bine-value:after worked perfectly.
